src/preamble.js

```diff
@@ -749,6 +752,50 @@ function instrumentWasmTableWithAbort() {
 }
 #endif
 
+#if MEMORY64
+// In memory64 mode wasm pointers are 64-bit. To support that in JS we must use
+// BigInts. For now we keep JS as much the same as it always was, that is,
+// stackAlloc() receives and returns a Number from the JS point of view -
+// we translate BigInts automatically for that.
+function instrumentWasmExportsForMemory64(exports) {
+  var instExports = {};
+  for (var name in exports) {
+    (function(name) {
+      var original = exports[name];
+      var replacement = original;
+      if (name === 'stackAlloc' || name === 'malloc') {
```

Comment:
This method won't work if we minify the names of exports, which we do in `-O3` etc. For now, please add a comment `// TODO: support minified export names`, and in a later PR we should disable that minification for `MEMORY64` for now (that would be in `minify_wasm_imports_and_exports` in `tools/building.py`)

Reply:
Ok, will do.

Reply:
Addressed here: https://github.com/emscripten-core/emscripten/pull/15305